tensorflow/python/framework/tensor_util_test.py

```diff
@@ -662,6 +663,69 @@ def testShapeEquals(self):
     self.assertFalse(tensor_util.ShapeEquals(t, [1, 4]))
     self.assertFalse(tensor_util.ShapeEquals(t, [4]))
 
+  def testMockArray(self):
+    class MockArray(object):
+      def __init__(self, array):
+        self.array = array
+
+      def __array__(self, dtype=None):
+        return np.asarray(self.array, dtype)
+
+    ma = MockArray(np.array([10, 20, 30]))
+    t = tensor_util.make_tensor_proto(ma)
+    if sys.byteorder == "big":  
```

Comment:
Actually, instead of using tensor proto, which relies on big vs little endian, how about just session.run the return value of ops.convert_to_tensor and comparing with the original .__array__ result?